src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/SdkClient/ResourceManagerSdkClient.cs

```diff
@@ -306,6 +306,33 @@ public static List<string> ParseDetailErrorMessage(string statusMessage)
             return null;
         }
 
+        public static List<Tuple<string, string>> ParseDetailErrorMessageWithCode(string statusMessage)
```

Comment:
Tuples are generally code smell. Can this be a shared complex type?
{
  errorCode: ""
  errorMessage: ""
}

Reply:
I actually see us (ARM) using Tuples throughout our code. Is there a particular reason (perf impact?) to avoid Tuples? Converting this to its own type is straightforward, but unless there is a strong need to do that, I'd prefer to keep it as-is. We can also take this as a separate change (there is a need for another PR to update ALL our ARM cmdlets to start using async header either way) if needed